lib/resque/worker.rb

```diff
@@ -103,7 +103,7 @@ def self.find(worker_id, options = {})
       skip_exists = options[:skip_exists]
 
       if skip_exists || exists?(worker_id)
-        host, pid, queues_raw = worker_id.split(':')
+        host, pid, queues_raw = worker_id.split(':', 3)
```

Comment:
Why have you chosen 3 as the limit?

Reply:
I believe this is because the worker id is structured as `host:pid:queue_name` where `queue_name` could include any number of colons. Limiting it to 3 ensures we get 3-element array that we want.